src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -284,7 +284,7 @@ internal static BindingRestrictions CombineRestrictions(this DynamicMetaObject t
             var targetTemp = Expression.Variable(target.Expression.Type);
```

Comment:
@vexx32, your last commit had 2 failures in `PowerShell-CI-windows`
Enter-PSHostProcess tests.By Process Id.Can enter using NamedPipeConnectionInfo
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Enter-PSHostProcess.Tests.ps1: line 148
148:                     Should -BeTrue -Because (get-content $splat.FilePath -Raw)
```
TabCompletion.Miscellaneous completion tests.Test history completion
```powershell
Expected strings to be the same, but they were different.
Expected length: 23
Actual length:   633
Strings differ at index 0.
Expected: 'Test history completion'
But was:  '$env:POWER...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\TabCompletion\TabCompletion.Tests.ps1: line 909
909:             $res.CompletionMatches[0].CompletionText | Should -BeExactly "Test history completion"
```